test/integration/conftest.py

```diff
@@ -50,3 +53,21 @@ def container_runtime_installed():
         except FileNotFoundError:
             pass
     pytest.skip('No container runtime is available.')
+
+
+@pytest.fixture(scope='session')
+def clear_integration_artifacts(request):
+    '''Fixture is session scoped to allow parallel runs without error
+    '''
+    if 'PYTEST_XDIST_WORKER' in os.environ:
+        # we never want to clean artifacts if running parallel tests
+        # because we cannot know when all processes are finished and it is
+        # safe to clean up
+        return
+
+    def rm_integration_artifacts():
+        path = "test/integration/artifacts"
```

Comment:
This isn't totally coherent, because this isn't the only private_data_dir in the tests. We don't bother cleaning up the artifacts from the other directories. Obviously I have preferences - I have been adding new private_data_dirs to `test/data` folder, because I don't want them polluting the test files. I get that someone else looked at this and also said "hey, I don't want these artifact files polluting the test files". So I get it, but this was a partial answer which is arguably worse than no answer. It complicated unrelated efforts to parallelize the tests, and the problem quickly re-emerged in `test/integration/containerized/priv_data`, if you could have called it solved for `test/integration/artifacts` in the first place.

My goal here is not to refactor, and thus I try to make this only slightly less bad. My goal here is to un-break the multiprocessing.